### ISSUE TITLE
"Enable payment button" and "Enable donation form" links redirect to an unexpected place

### ISSUE THREAD

**Issue by foleynotrose** [Automattic/jetpack#35393]
### Impacted plugin

Jetpack

### Quick summary

When in the Jetpack > Settings > Monetize dashboard, the `Enable payment button` and `Enable donation form` links redirect to an unexpected place. 

### Steps to reproduce

1. On an ads-enabled Jetpack site, go to Jetpack > Settings > Monetize. 
2. Scroll to Collect Payments and click `Enable Payment Button` OR 
3. Scroll to Accept donation and tips and click `Enable donations form.` 

### A clear and concise description of what you expected to happen.

Each link should go to a page specific to that setting, or maybe the main `Monetization Options` tab. 

### What actually happened

Redirect goes to the `Payment Settings` tab (https://wordpress.com/earn/payments/URL), which is not related to either of above mentioned options. Might make the user feel lost. 

<img width="1126" alt="Screenshot 2024-02-01 at 14 41 42" src="https://github.com/Automattic/jetpack/assets/26651984/ccf06e7d-8f8a-4291-ae0d-ec1a47dc9dc7">

### Impact

Most (> 50%)

### Available workarounds?

Yes, difficult to implement

### Platform (Simple and/or Atomic)

Self-hosted

**Comment by jeherve** [comment]
> Each link should go to a page specific to that setting, or maybe the main Monetization Options tab.

I worry the main Monetization Options tab would just feel like sending people in circles, from one page to another. Do you have a specific idea of a page where we could send them instead? How about the support docs for each block? Would that be better?

https://jetpack.com/support/jetpack-blocks/payments-block/
https://jetpack.com/support/jetpack-blocks/donations-block/

**Comment by foleynotrose** [comment]
@jeherve yes the support docs about those features would be better, but in that case it would probably be better if the link said `Learn more`. 

If i'm clicking a link that says "enable" something, I expect to be taken to the place where I can actually enable that thing.

**Comment by jeherve** [comment]
cc @simison since you worked with me on #31696. What's your take on this suggestion? Should I update to "Learn more"? Should I remove the entire CTA, since there is already a link to the support doc on the right of that card?
